### PR TITLE
[synapse-artifacts] release for Octorber

### PR DESCRIPTION
**Pull request by kevinzz6**
# Description
swagger definition:
https://github.com/Azure/azure-rest-api-specs/tree/main/specification/synapse/data-plane

swagger SHA:
5ae522bc106bf8609c6cb379e584aa3e0e2639f3
# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by kevinzz6** [comment]
cc @msyyc  @wonner  for awareness

File: sdk/synapse/azure-synapse-artifacts/azure/synapse/artifacts/_version.py [hunk 1]
```diff
@@ -6,4 +6,4 @@
 # Changes may cause incorrect behavior and will be lost if the code is regenerated.
 # --------------------------------------------------------------------------
 
-VERSION = "0.17.1"
+VERSION = "0.17.0"
```
Comment (wonner) [inline sdk/synapse/azure-synapse-artifacts/azure/synapse/artifacts/_version.py:9]: 0.18.0?

Comment (kevinzz6) [reply sdk/synapse/azure-synapse-artifacts/azure/synapse/artifacts/_version.py:9]: yes, changge to 0.18

File: sdk/synapse/azure-synapse-artifacts/azure/synapse/__init__.py [hunk 2]
```diff
@@ -1 +1 @@
-__path__ = __import__("pkgutil").extend_path(__path__, __name__)
+__path__ = __import__("pkgutil").extend_path(__path__, __name__)  # type: ignore
```
Comment (msyyc) [inline sdk/synapse/azure-synapse-artifacts/azure/synapse/__init__.py:1]: It is better to add some test case

Comment (msyyc) [reply sdk/synapse/azure-synapse-artifacts/azure/synapse/__init__.py:1]: You can refer to guidance https://github.com/Azure/azure-sdk-for-python/wiki/Dataplane-Codegen-Quick-Start-for-Test and existing testcase https://github.com/Azure/azure-sdk-for-python/tree/e0c7db941d9f512bab01a75c4a43d26cbfe946f2/sdk/synapse/azure-synapse-artifacts/tests.

**Review by msyyc — APPROVED** [review]

File: sdk/synapse/azure-synapse-artifacts/azure/synapse/__init__.py (hunk 2 above)
Comment (kevinzz6) [reply sdk/synapse/azure-synapse-artifacts/azure/synapse/__init__.py:1]: Sure. Will add test case on next release.

**Pull request merged**